lib/twitter_cldr/utils/range_helpers.rb

```diff
@@ -0,0 +1,48 @@
+# encoding: UTF-8
+
+# Copyright 2012 Twitter, Inc
+# http://www.apache.org/licenses/LICENSE-2.0
+
+require 'set'
+
+module TwitterCldr
+  module Utils
+
+    module RangeHelpers
+      def overlap?(range1, range2)
+        is_numeric_range?(range1) && is_numeric_range?(range2) && (
+          front_overlap?(range1, range2) ||
```

Comment:
Do we ever use these distinct overlap cases? If not, can we replace these three conditions with [this implementation](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/range/overlaps.rb)?